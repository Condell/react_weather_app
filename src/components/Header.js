import React from "react";
import styled from "styled-components";

import Input from "./shared/Input";
import Button from "./shared/Button";

const HeaderWrapper = styled.div`
  background-color: #f8642e;
  height: 20vh;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
  align-content: center;
`;

const Title = styled.h1`
  font-family: 'Abel', sans-serif;
  margin: 0px;
  color: white;
`;

const SmallButton = styled(Button)`
    padding: .5em;
`;

const Header = () =>
  <HeaderWrapper>
    <Title>Weather App</Title>
    <Input type="text" placeholder="St. George, Utah" size="14" />
    <SmallButton>Get Weather</SmallButton>
  </HeaderWrapper>;

export default Header;
