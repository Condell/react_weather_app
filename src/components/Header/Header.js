import React from "react";
import styled from "styled-components";

import WeatherFormContainer from "../WeatherFormContainer/WeatherFormContainer";
import Title from "../Title/Title";

const HeaderWrapper = styled.div`
  background-color: #f8642e;
  min-height: 20vh;
  display: flex;
  flex-flow: column;
  align-items: center;
  justify-content: space-around;
`;

const Header = () =>
  <HeaderWrapper>
    <Title whiteFont>Weather App</Title>
    <WeatherFormContainer row />
  </HeaderWrapper>;

export default Header;
