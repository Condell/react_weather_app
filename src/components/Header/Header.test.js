import React from "react";
import { shallow } from "enzyme";
import 'jest-styled-components'
import toJson from "enzyme-to-json"

import Header from "./Header";


describe("Header", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Header />);
  });

  it("should render", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
