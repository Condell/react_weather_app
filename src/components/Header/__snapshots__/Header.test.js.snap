// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render 1`] = `
<styled.div>
  <styled.h1
    whiteFont={true}
  >
    Weather App
  </styled.h1>
  <WeatherFormContainer
    row={true}
  />
</styled.div>
`;
